src/commands/PublishCommand.js

```diff
@@ -419,7 +425,12 @@ export default class PublishCommand extends Command {
         version: update.package.version,
         location: update.package.location
       };
-      const recommendedVersion = recommendVersionFn(pkg, this.execOpts);
+
+      const changelogOptions = Object.assign({}, this.execOpts, {
```

Comment:
Let's make this a lazy getter, so we only have to merge the options once:

```patch
diff --git a/src/commands/PublishCommand.js b/src/commands/PublishCommand.js
index 9edf8ad5..65222478 100644
--- a/src/commands/PublishCommand.js
+++ b/src/commands/PublishCommand.js
@@ -152,6 +152,20 @@ export default class PublishCommand extends Command {
     });
   }
 
+  get changelogOpts() {
+    if (!this._changelogOpts) {
+      const {
+        changelogPreset,
+      } = this.options;
+
+      this._changelogOpts = Object.assign({}, this.execOpts, {
+        changelogPreset,
+      });
+    }
+
+    return this._changelogOpts;
+  }
+
   initialize(callback) {
     this.gitRemote = this.options.gitRemote || "origin";
     this.gitEnabled = !(this.options.canary || this.options.skipGit);
```

Then just pass `this.changelogOpts` to the various functions.